clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -915,25 +915,25 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
 
   ModulePassManager MPM;
 
-  // FIXME: Change this when -fno-sycl-early-optimizations is not tied to
-  // -disable-llvm-passes.
-  if (CodeGenOpts.DisableLLVMPasses && LangOpts.SYCLIsDevice)
+  if (LangOpts.SYCLIsDevice && !CodeGenOpts.DisableSYCLEarlyOpts)
```

Comment:
Lets' move this under `if (!CodeGenOpts.DisableLLVMPasses) {`, so `-disable-llvm-passes` works correctly for SYCL passes as well

Reply:
This change causes the following tests to fail:

```
Failed Tests (3):
  Clang :: CodeGenSYCL/device_has.cpp
  Clang :: CodeGenSYCL/sub-group-size.cpp
  Clang :: CodeGenSYCL/uses_aspects.cpp
```

Reply:
I see that at least `device_has` uses `-disable-llvm-passes` and I would say that this is a bug in the test. The proper behavior of `-disable-llvm-passes` is to disable all passes.

Perhaps we need to refactor those test to outline the part of them which relies on the pass to leave the rest to test CodeGen component of clang as expected.

Tagging @steffenlarsen and @KornevNikita for their feedback

Reply:
Won't disabling the aspect propagation pass affect functionality at O0?

Reply:
'-O0' does not disables passes, nor it disables early optimisations pipeline (even though it will be almost empty), but anyway aspects propagation pass is inserted regardless of the optimisation level